source/validate_id.cpp

```diff
@@ -2002,11 +2002,54 @@ bool idUsage::isValid<OpBranch>(const spv_instruction_t *inst,
                                 const spv_opcode_desc opcodeEntry) {}
 #endif
 
-#if 0
 template <>
-bool idUsage::isValid<OpBranchConditional>(
-    const spv_instruction_t *inst, const spv_opcode_desc opcodeEntry) {}
-#endif
+bool idUsage::isValid<SpvOpBranchConditional>(
+    const spv_instruction_t *inst, const spv_opcode_desc opcodeEntry)
+{
+  const size_t numOperands = inst->words.size() - 1;
+  constexpr size_t condOperandIndex = 1;
+  constexpr size_t targetTrueIndex = 2;
+  constexpr size_t targetFalseIndex = 3;
+  constexpr size_t weightTrueIndex = 4;
+  constexpr size_t weightFalseIndex = 5;
+
+  // num_operands is either 3 or 5 --- if 5, the last two need to be literal integers
```

Comment:
I'm not actually sure how to validate that the last two operands are literal integers. It seems like they're just two dwords in the instruction that can always be interpreted as literal integers.

Is there something that we need to check here?

Reply:
I don't think so.

If you write code directly in SPIR-V (i.e. not in assembly, but in raw SPIR-V) and you put here an id referencing the integer you want to use instead of a literal integer, then there is pretty much nothing we can do: the id would be interpreted as the weight and the code would look perfectly valid.